cmd/apiserver/main.go

```diff
@@ -156,7 +156,22 @@ func main() {
 			ClientAuth:         tls.VerifyClientCertIfGiven,
```

Comment:
The [API documentation](https://access.crunchydata.com/documentation/postgres-operator/4.6.1/advanced/direct-api-calls/) states that client certs and keys are required, but the config here doesn't seem to agree.

Should [`ClientAuth`](https://golang.org/pkg/crypto/tls/#ClientAuthType) be set to `RequireAndVerifyClientCert`? Maybe it's also necessary to check if, this needs to be a relaxed option if `tlsNoVerify` is `true`, or if `InsecureSkipVerify` will handle all that..

Reply:
Hm. The usage of that setting in this context slightly predates me. Let me dig up the history as to why it's set that way.

If possible, I agree with the approach of:

```go
clientAuth := tls. RequireAndVerifyClientCert
if tlsNoVerify {
	clientAuth = tls.InsecureSkipVerify
}
```

for a plethora of reasons which I'm fairly confident you'd agree with.

Reply:
Discussing with a few others, I think the item that requires research / testing around this is the effect on the health checkpoints, `/health` and `/healthz`. `RequireAndVerify...` may preclude those from working, so it will take a bit more research and testing.

Reply:
Proceeding without this change, based on the health check considerations. Perhaps it can be strengthened if/when support for customisation of these config parameters is added.